manager/controlapi/secret.go

```diff
@@ -245,9 +242,11 @@ func validateSecretSpec(spec *api.SecretSpec) error {
 	if err := validateConfigOrSecretAnnotations(spec.Annotations); err != nil {
 		return err
 	}
-
-	if len(spec.Data) >= MaxSecretSize || len(spec.Data) < 1 {
-		return grpc.Errorf(codes.InvalidArgument, "secret data must be larger than 0 and less than %d bytes", MaxSecretSize)
+	if spec.Driver.Name != "" {
+		return nil
+	}
+	if err := validation.ValidateSecretPayload(spec.Data); err != nil {
+		return grpc.Errorf(codes.InvalidArgument, err.Error())
```

Comment:
`grpc.Errorf(codes.InvalidArgument, "%s", err.Error())`, in case the error string has `%` characters inside it.